text/0000-contingency-plan.md

```diff
@@ -0,0 +1,425 @@
+minimize the impact on the ecosystem.
+
+The first step then is to evaluate the impact of the fix on the crates
+found in the `crates.io` website (using e.g. the crater tool). If
+impact is found to be "small" (which this RFC does not attempt to
+precisely define), then the fix can simply be landed. As today, the
+commit message of any breaking change should include the term
+`[breaking-change]` along with a description of how to resolve the
+problem, which helps those people who are affected to migrate their
+code. A description of the problem should also appear in the relevant
+subteam report.
+
+In cases where the impact seems larger, the following steps can be
+taken to ease the transition:
+
+1. Identify important crates (such as those with many dependencies)
+   and work with the crate author to correct the code as quickly as
+   possible, ideally before the fix even lands.
+2. Work hard to ensure that the error message identifies the problem
+   clearly and suggests the appropriate solution.
+3. Provide an annotation that allows for a scoped "opt out" of the
+   newer rules, as described below. While the change is still
+   breaking, this at least makes it easy for crates to update and get
+   back to compiling status quickly.
+4. Begin with a deprecation or other warning before issuing a hard
+   error. In extreme cases, it might be nice to begin by issuing a
+   deprecation warning for the unsound behavior, and only make the
+   behavior a hard error after the deprecation has had time to
+   circulate. This gives people more time to update their crates.
+   However, this option may frequently not be available, because the
+   source of a compilation error is often hard to pin down with
+    precision.
+   
+Some of the factors that should be taken into consideration when
+deciding whether and how to minimize the impact of a fix:
+
+- How many crates on `crates.io` are affected?
+  - This is a general proxy for the overall impact (since of course
+    there will always be private crates that are not part of
+    crates.io).
+- Were particularly vital or widely used crates affected?
+  - This could indicate that the impact will be wider than the raw
+    number would suggest.
+- Does the change silently change the result of running the program,
+  or simply cause additional compilation failures?
+  - The latter, while frustrating, are easier to diagnose.
+- What changes are needed to get code compiling again? Are those
+  changes obvious from the error message?
+  - The more cryptic the error, the more frustrating it is when
+    compilation fails.
+    
+#### Opting out
+
+In some cases, it may be useful to permit users to opt out of new type
+rules. The intention is that this "opt out" is used as a temporary
+crutch to make it easy to get the code up and running. Depending on
+the severity of the soundness fix, the "opt out" may be permanently
+available, or it could be removed in a later release. In either case,
```

Comment:
@nrc 

> A "permanently available" soundness hole sounds very bad to me. 

I agree, I was envisioning primarily "less dangerous" kinds of soundness holds, such as bugs in coherence that could lead to overlapping impls. I'll rework the text though to make it clear that "permanently available" changes would be the exception.